src/components/sbb-expansion-panel/sbb-expansion-panel.tsx

```diff
@@ -0,0 +1,213 @@
+import {
+  Component,
+  ComponentInterface,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  JSX,
+  Listen,
+  Prop,
+  Watch,
+} from '@stencil/core';
+import { InterfaceTitleAttributes } from '../sbb-title/sbb-title.custom';
+import { toggleDatasetEntry } from '../../global/dom';
+import { InterfaceSbbExpansionPanelAttributes } from './sbb-expansion-panel.custom';
+
+let nextId = 0;
+
+/**
+ * @slot header - Use this to render the sbb-expansion-panel-header.
+ * @slot content - Use this to render the sbb-expansion-panel-content.
+ */
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-expansion-panel.scss',
+  tag: 'sbb-expansion-panel',
+})
+export class SbbExpansionPanel implements ComponentInterface {
+  /** Heading level; if unset, a `div` will be rendered. */
+  @Prop() public titleLevel?: InterfaceTitleAttributes['level'];
+
+  /** The background color of the panel. */
+  @Prop() public color: InterfaceSbbExpansionPanelAttributes['color'] = 'white';
+
+  /** Whether the panel is expanded. */
+  @Prop({ mutable: true, reflect: true }) public expanded = false;
+
+  /** Whether the panel is disabled, so its expanded state can't be changed. */
+  @Prop({ reflect: true }) public disabled = false;
+
+  /** Whether the panel has no border. */
+  @Prop({ reflect: true }) public borderless = false;
+
+  /** Whether the animations should be disabled. */
+  @Prop({ reflect: true }) public disableAnimation = false;
+
+  /** Emits whenever the sbb-expansion-panel starts the opening transition. */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'will-open',
+  })
+  public willOpen: EventEmitter<void>;
+
+  /** Emits whenever the sbb-expansion-panel is opened. */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'did-open',
+  })
+  public didOpen: EventEmitter<void>;
+
+  /** Emits whenever the sbb-expansion-panel begins the closing transition. */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'will-close',
+  })
+  public willClose: EventEmitter<void>;
+
+  /** Emits whenever the sbb-expansion-panel is closed. */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'did-close',
+  })
+  public didClose: EventEmitter<void>;
+
+  @Element() private _element: HTMLElement;
+
+  @Listen('toggle-expanded')
+  public toggleExpanded(): void {
+    this.expanded = !this.expanded;
+  }
+
+  @Watch('expanded')
+  public onExpandedChange(): void {
+    this._element
+      .querySelector('sbb-expansion-panel-header')
+      .setAttribute('aria-expanded', String(this.expanded));
+    this._element
+      .querySelector('sbb-expansion-panel-content')
+      .setAttribute('aria-hidden', String(!this.expanded));
```

Comment:
Could _headerRef and _contentRef be used here?

Reply:
Yes, i did it in [23c4d25](https://github.com/lyne-design-system/lyne-components/pull/1860/commits/23c4d2557ee5eae7ed464f41ad211275b467056a)